include/internal/sockets.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 1995-2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+
+#ifndef HEADER_OPENSSL_SOCKETS
+
+# ifdef OPENSSL_NO_SOCK
+# elif defined(WINDOWS) || defined(MSDOS)
```

Comment:
Replace all occurences of `WINDOWS` and `MSDOS` with `OPENSSL_SYS_WINDOWS` and `OPENSSL_SYS_MSDOS`.  If you have a look in `e_os.h`, you'll find that `WINDOWS` and `MSDOS` are defined there, but obviously not here, which explains the Appveyor failures.

Also, you might want to consider including `<openssl/e_os2.h>` in this file.